modules/ROOT/pages/dataweave-streaming.adoc

```diff
@@ -0,0 +1,398 @@
+        <customer number="123123">Joe</customer>
+    </header>
+    <order-items>
+        <order-item id="31">
+            <product>111</product>
+            <quantity>2</quantity>
+            <price>8.90</price>
+        </order-item>
+        <order-item id="31">
+            <product>222</product>
+            <quantity>7</quantity>
+            <price>5.20</price>
+        </order-item>
+        <order-item id="31">
+            <product>111</product>
+            <quantity>2</quantity>
+            <price>8.90</price>
+        </order-item>
+        <order-item id="31">
+            <product>222</product>
+            <quantity>7</quantity>
+            <price>5.20</price>
+        </order-item>
+        <order-item id="31">
+            <product>222</product>
+            <quantity>7</quantity>
+            <price>5.20</price>
+        </order-item>
+    </order-items>
+</order>
+----
+​
+Given this XML source data, you can set unit of the stream as `<order-item/>` by setting `collectionPath='order.order-items'` in the `outputMimeType` value, for example:
+
+<flow name="dw-streaming-example" >
+  <http:listener doc:name="Listener"
+      outputMimeType="application/xml; collectionpath='order.order-items'; streaming=true"
+      config-ref="HTTP_Listener_config" path="/input"/>
+</flow>
+
+The following DataWeave script is able to stream the XML input using each `<order-items/>` element as the streamable unit.
+
+[source,DataWeave,linenums]
+----
+%dw 2.0
+output application/xml
+---
+{
+  salesorder: {
+    itemList: payload.order."order-items".*"order-item" map {
+      ("i_" ++ $$) : {
+        id: $.@id,
+        productId: $.product,
+        quantity: $.quantity,
+        price: $.price
+      }
+    }
+  }
+}
+----
+
+The script produces the following XML output.
+
+[source,XML,linenums]
+----
+<?xml version='1.0' encoding='UTF-8'?>
+<salesorder>
+  <itemList>
+    <i_0>
+      <id>31</id>
+      <quantity>2</quantity>
+      <productId>111</productId>
+      <price>8.90</price>
+    </i_0>
+  </itemList>
+  <itemList>
+    <i_1>
+      <id>31</id>
+      <quantity>7</quantity>
+      <productId>222</productId>
+      <price>5.20</price>
+    </i_1>
+  </itemList>
+  <itemList>
+    <i_2>
+      <id>31</id>
+      <quantity>2</quantity>
+      <productId>111</productId>
+      <price>8.90</price>
+    </i_2>
+  </itemList>
+  <itemList>
+    <i_3>
+      <id>31</id>
+      <quantity>7</quantity>
+      <productId>222</productId>
+      <price>5.20</price>
+    </i_3>
+  </itemList>
+  <itemList>
+    <i_4>
+      <id>31</id>
+      <quantity>7</quantity>
+      <productId>222</productId>
+      <price>5.20</price>
+    </i_4>
+  </itemList>
+</salesorder>
+----
+​
+== Validate that a Script is Streamable (Experimental Feature)
+​
+To help you code in a streamable way and avoid mistakes that can break the streaming process, DataWeave provides an _advanced, experimental_ annotation and a related directive:
```

Comment:
```suggestion
To help you code so that streaming is successful, DataWeave provides the following _advanced, experimental_ annotation and a related directive:
```